samples/subsys/zbus/hello_world/src/main.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+struct version_msg {
+	uint8_t major;
+	uint8_t minor;
+	uint16_t build;
+};
+
+struct acc_msg {
+	int x;
+	int y;
+	int z;
+};
+
+ZBUS_CHAN_DEFINE(version_chan,	     /* Name */
+		 true,		     /* Read only */
+		 struct version_msg, /* Message type */
+
+		 NULL,		       /* Validator */
+		 ZBUS_OBSERVERS_EMPTY, /* observers */
+		 ZBUS_MSG_INIT(.major = 0, .minor = 1,
+			       .build = 2) /* Initial value major 0, minor 1, build 2 */
+);
+
+ZBUS_CHAN_DEFINE(acc_data_chan,	 /* Name */
+		 false,		 /* Read only */
+		 struct acc_msg, /* Message type */
+
+		 NULL,				       /* Validator */
+		 ZBUS_OBSERVERS(foo_lis, bar_sub),     /* observers */
+		 ZBUS_MSG_INIT(.x = 0, .y = 0, .z = 0) /* Initial value */
+);
+
+bool simple_chan_validator(void *msg, size_t msg_size)
+{
+	ARG_UNUSED(msg_size);
+
+	int *simple = msg;
+
+	if ((*simple >= 0) && (*simple < 10)) {
+		return true;
+	}
+
+	return false;
+}
+
+ZBUS_CHAN_DEFINE(simple_chan, /* Name */
+		 false,	      /* Read only */
+		 int,	      /* Message type */
+
+		 simple_chan_validator, /* Validator */
+		 ZBUS_OBSERVERS_EMPTY,	/* observers */
+		 0			/* Initial value is 0 */
+);
+
+void listener_callback_example(struct zbus_channel *chan)
+{
+	struct acc_msg *acc = zbus_chan_msg(chan);
+
+	LOG_INF("From listener -> Acc x=%d, y=%d, z=%d", acc->x, acc->y, acc->z);
+}
+
+ZBUS_LISTENER_DEFINE(foo_lis, listener_callback_example);
+
+ZBUS_SUBSCRIBER_DEFINE(bar_sub, 4);
+
+void subscriber_task(void)
+{
+	struct zbus_channel *chan = NULL;
+
+	while (!zbus_sub_wait(&bar_sub, &chan, K_FOREVER)) {
+		struct acc_msg acc = {0};
+
+		if (&acc_data_chan == chan) {
+			zbus_chan_read(&acc_data_chan, &acc, K_MSEC(500));
+
+			LOG_INF("From subscriber -> Acc x=%d, y=%d, z=%d", acc.x, acc.y, acc.z);
+		}
+	}
+}
+
+K_THREAD_DEFINE(subscriber_task_id, 512, subscriber_task, NULL, NULL, NULL, 3, 0, 0);
+
+int count;
+
+bool print_channel_data_iterator(struct zbus_channel *chan)
+{
+	LOG_INF("%d - Channel %s:", count, chan->name);
+	LOG_INF("      Message size: %d", chan->message_size);
+	++count;
+	LOG_INF("      Observers:");
+	for (struct zbus_observer **obs = chan->observers; *obs != NULL; ++obs) {
+		LOG_INF("      - %s", (*obs)->name);
+	}
+
+	return true;
+}
+
+bool print_observer_data_iterator(struct zbus_observer *obs)
+{
+	LOG_INF("%d - %s %s", count, ((obs->queue != NULL) ? "Subscriber" : "Listener"), obs->name);
```

Comment:
Drop the parenthesis from the ternary? (not sure it's a guideline but, you are putting double parenthesis here and no parenthesis on the one of 156) Personaly I would drop the `!= NULL` as well.

`LOG_INF("%d - %s %s", count, obs->queue ? "Subscriber" : "Listener", obs->name);` <- should work fine and is more compact